gdx/src/com/badlogic/gdx/maps/tiled/BaseTmjMapLoader.java

```diff
@@ -690,98 +693,94 @@ else if (compression.equals("zlib"))
 		return ids;
 	}
 
-	protected void loadTileSet (JsonValue element, FileHandle tmjFile, ImageResolver imageResolver) {
-		if (element.getString("firstgid") != null) {
-			int firstgid = element.getInt("firstgid", 1);
-			String imageSource = "";
-			int imageWidth = 0;
-			int imageHeight = 0;
-			FileHandle image = null;
-
-			String source = element.getString("source", null);
-			if (source != null) {
-				FileHandle tsj = getRelativeFileHandle(tmjFile, source);
-				try {
-					element = json.parse(tsj);
-					if (element.has("image")) {
-						imageSource = element.getString("image");
-						imageWidth = element.getInt("imagewidth", 0);
-						imageHeight = element.getInt("imageheight", 0);
-						image = getRelativeFileHandle(tsj, imageSource);
-					}
-				} catch (SerializationException e) {
-					throw new GdxRuntimeException("Error parsing external tileSet.");
-				}
-			} else {
-				if (element.has("image")) {
-					imageSource = element.getString("image");
-					imageWidth = element.getInt("imagewidth", 0);
-					imageHeight = element.getInt("imageheight", 0);
-					image = getRelativeFileHandle(tmjFile, imageSource);
-				}
-			}
-			String name = element.getString("name", null);
-			int tilewidth = element.getInt("tilewidth", 0);
-			int tileheight = element.getInt("tileheight", 0);
-			int spacing = element.getInt("spacing", 0);
-			int margin = element.getInt("margin", 0);
-
-			JsonValue offset = element.get("tileoffset");
-			int offsetX = 0;
-			int offsetY = 0;
-			if (offset != null) {
-				offsetX = offset.getInt("x", 0);
-				offsetY = offset.getInt("y", 0);
-			}
-			TiledMapTileSet tileSet = new TiledMapTileSet();
+	public TiledMapTileSet loadTileSet (JsonValue rawElement, FileHandle tmjFile, ImageResolver imageResolver) {
```

Comment:
Normally docs on this would be important, but I'm inclined to be lenient because it's an AssetManager Loader and those seem to be less critical for docs. I think anyone who needs to be really digging into AssetManager problems should be looking at the source code anyway. I'm a little unclear why the JsonValue is `raw` and what that means here. I think I could probably figure it out by how it's used, assuming it's used somewhere in the code, though it doesn't look like there's a new test in this PR.

Reply:
Also, the capitalization is different here too, `TileSet` should be `Tileset` to match the rest of the code. I'm not sure if this is a safe change to make, though, because it was TileSet before in this method only. Ugh... We should keep this as it is unless we know nothing relies on the old name...